### PR TITLE
first pass at install

### PR DESCRIPTION
**Pull request by netskink**
Output for this patch when doing `zopen build`

```
<stuff snipped>
:DEBUG:: 'Cleaning temp resources.'
:DEBUG:: 'Switching to previous cwd - current work dir was purged.'
- Integration complete.
- Sourcing environment to run any setup
CWD:: ./.appenv 2: .: ./.env 28: .: ./setup.sh 3: FSUM7351 not found
/z/jd895801/zopen/dev/depot_toolsport/depot_tools: ./.appenv 3: .: ./.env 28: .: ./setup.sh 3: FSUM9209 cannot execute: reason code = 0b1b011f: EDC5111I Permission denied.
Setting up depot_tools...
***** append_to_setup() start
***** append_to_setup() start
***** append_to_setup() start
CWD:
/z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main
ZOPEN_INSTALL_DIR: /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main
Creating a python venv...
Activating python venv named venv...
source: ./setup.sh 16: FSUM7351 not found
Upgrading pip in venv...
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: pip in /z/jd895801/.local/lib/python3.11/site-packages (24.0)
Installing packages via requirements.txt...
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: httplib2 in /z/jd895801/.local/lib/python3.11/site-packages (from -r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 1)) (0.22.0)
Requirement already satisfied: six in /usr/lpp/IBM/cyp/v3r11/pyz/lib/python3.11/site-packages (from -r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 2)) (1.16.0)
Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /z/jd895801/.local/lib/python3.11/site-packages (from httplib2->-r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 1)) (3.1.1)
***** append_to_setup() end
***** append_to_setup() end
***** append_to_setup() end
Setup completed.
zopen-build completed in 100 seconds.
```

### CODE REVIEW

File: buildenv [hunk 1]
```diff
@@ -31,11 +31,82 @@ zopen_check_results()
...
+  # usr/local/zopen dir
+
+  echo "***** in append_to_env() start"
+  echo "CWD: "
+  pwd
+
+
+
+  # update the path for depot_tools
+  export PATH=${PATH}:${HOME}/zopen/dev/depot_toolsport/depot_tools
```
Comment (IgorTodorovskiIBM) [inline buildenv]: zopen build automatically adds the install path + "/bin" to the PATH envrionment variable

File: buildenv [hunk 2]
```diff
@@ -31,11 +31,82 @@ zopen_check_results()
...
+  # update the path for pip
+  export PATH=${PATH}:${HOME}/.local/bin
+  # update the compiler settings
+  # TODO: this can probably be moved to v8 build later
+  export CC=clang
+  export CXX=clang++
+  export LINK=clang++
+  # Disable updating depot_tools
+  export DEPOT_TOOLS_UPDATE=0
```
Comment (IgorTodorovskiIBM) [inline buildenv]: For the exports to show up, you'll need to echo them:
```
echo "export DEPOT_TOOLS_UPDATE=0"
```

File: buildenv [hunk 3]
```diff
@@ -31,11 +31,82 @@ zopen_check_results()
...
+echo "***** append_to_setup() start"
+echo "***** append_to_setup() start"
+echo "***** append_to_setup() start"
+echo "CWD:"
+pwd
+# /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main
+echo "ZOPEN_INSTALL_DIR: $ZOPEN_INSTALL_DIR"
+# /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main
+echo "Creating a python venv..."
+python3 -m venv venv
```
Comment (IgorTodorovskiIBM) [inline buildenv]: Same for this, for "python3 -m venv venv" to show up in the setup.sh script, you'll need to echo it:
```
echo "python3 -m venv venv"
```

**Comment by netskink** [comment]
Removing the `here` document results in a better looking ~/zopen/usr/local/zopen/depot_tools/depot_tools/setup.sh file.

Here is the contents of the file:

```
#!/bin/sh
touch ".installing"
. ./.env
echo "Setting up depot_tools..."
echo "  Creating a python venv..."
python3 -m venv venv
echo "  Activating python venv named venv..."
source /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/venv/bin/activate
echo "  Upgrading pip in venv..."
python -m pip install -U pip
echo "Installing packages via requirements.txt..."
pip install -r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt
echo "Setup completed."
touch ".installed"
rm ".installing"
```

Here is the output from zopen:

```
:DEBUG:: 'Cleaning temp resources.'
:DEBUG:: 'Switching to previous cwd - current work dir was purged.'
- Integration complete.
- Sourcing environment to run any setup
CWD:: ./.appenv 2: .: ./.env 28: .: ./setup.sh 3: FSUM7351 not found
/z/jd895801/zopen/dev/depot_toolsport/depot_tools: ./.appenv 3: .: ./.env 28: .: ./setup.sh 3: FSUM9209 cannot execute: reason code = 0b1b011f: EDC5111I Permission denied.
Setting up depot_tools...
	Creating a python venv...
	Activating python venv named venv...
source: ./setup.sh 8: FSUM7351 not found
	Upgrading pip in venv...
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: pip in /z/jd895801/.local/lib/python3.11/site-packages (24.0)
Installing packages via requirements.txt...
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: httplib2 in /z/jd895801/.local/lib/python3.11/site-packages (from -r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 1)) (0.22.0)
Requirement already satisfied: six in /usr/lpp/IBM/cyp/v3r11/pyz/lib/python3.11/site-packages (from -r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 2)) (1.16.0)
Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /z/jd895801/.local/lib/python3.11/site-packages (from httplib2->-r /z/jd895801/zopen/usr/local/zopen/depot_tools/depot_tools-main/requirements.txt (line 1)) (3.1.1)
Setup completed.
zopen-build completed in 107 seconds.
````

It looks like the build output above has some errors, but on the upside, the venv is created in same dir as the setup.sh.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**